cfgov/scripts/archive_events.py

```diff
@@ -23,21 +24,26 @@ def run():
         archived_events = EventArchivePage.objects.get(
             title__icontains='Archive')
 
-        if len(events.get_children()) > 1:
-            for child in events.get_children():
-                event = child.specific
-                if isinstance(event, EventPage):
-                    if event.end_dt:
-                        if event.end_dt < timezone.now():
-                            if event.can_move_to(archived_events):
+        live_event_pages = events.get_children().live().exact_type(EventPage)
+
+        if len(live_event_pages) > 1:
+            for event in live_event_pages:
+                event = event.specific
+                if event.end_dt:
+                    if event.end_dt < timezone.now():
+                        if event.can_move_to(archived_events):
+                            try:
+                                event.move(archived_events, pos='last-child')
+                            except ValidationError:
+                                iso_date = event.start_dt.date().isoformat()
+                                event.slug = event.slug + '-' + iso_date
```

Comment:
Just wondering if it would be more useful to know when the event was archived, as opposed to when its original start date was. If so, you could convey that with `event.slug + '-archived-' + datetime.date.today().isoformat()`

Reply:
I think that for the purposes of differentiating these URL slugs, it's more natural to indicate the date that the event occurred (or started).

Reply:
I can see the usefulness of having the start date in the slug. Maybe no one cares when it was archived.